### DISCUSSION TITLE
Export Audio dialog: availability of a control depends on a control below it

### DISCUSSION THREAD

**Discussion by DavidBailes** [audacity/audacity#5430] [General]
The availability of the custom mapping radio button, depends on the setting of the Export range radio buttons, which are lower in the dialog. (The custom mapping button is not available when the Export range radio buttons are set to multiple files.)
I have the expectation (maybe mistaken) than a user should be able to go through the controls in a dialog in normal reading order and adjust the settings as desired, without having to back track because a previous control has now been made available.

**Comment by petersampsonaudacity** [comment]
@DavidBailes @LWinterberg 

Maybe, instead, this begs the question: 
just why is **Custom Mapping** not available for **Multiple Files** export - only for **Entire Project** or  **Current selection** ?

**Reply by LWinterberg** [reply]
If you export track by track, custom mapping kinda doesn't do anything, does it?

**Comment by petersampsonaudacity** [comment]
> If you export track by track, custom mapping kinda doesn't do anything, does it?

Aye, a good point @LWinterberg 

But if you export label by label then surely is does (or could do).

But if you decide not to make it available for Multiple export then @DavidBailes has a **_very valid_** point about the flow order in the dialog.

**Comment by LWinterberg** [comment]
Fwiw, we have some more of these cases: If you set MP3 to 48k sample rate, then bitrate mode to constant and quality to 8 kbps, the sample rate goes down to 24k. In MP2, the sample rate and bitrate depends on the MPEG version used. And changing the format technically also changes the file name above it. 

You're definitely right in thinking that it's good to have a downwards flow in a dialog, and not have any changes to a value after you've set it. There's two more principles important in this context however:

* Ordering by necessity - When given the task "send me an MP3", you near-necessarily will engage with the name, folder and format options, so these go first and in that order. all the other stuff, virtue of being further down the list, feels unimportant and "it's probably fine to keep this default", which is true. 
* Grouping by subject - When an option spawns more options (as the format dropdown and export multiple does), the "child" options want to be directly underneath the originally selected option. 

These principles aren't goals in themselves, however: The end goal is to produce a dialog that is cognitively manageable. I think compared to the alternatives, having a flow that's a tad wonky in rare situations is preferable over having a solution that keeps the flow pristine, but breaks one of the other two principles:

![image](https://github.com/audacity/audacity/assets/87814144/c0d6b6fc-10d9-462d-9b02-01ecefa3dd5c)

**Reply by petersampsonaudacity** [reply]
> I think compared to the alternatives, having a flow that's a tad wonky in rare situations is preferable over having a solution that keeps the flow pristine, but breaks one of the other two principles

I am minded to agree

**Comment by DavidBailes** [comment]
> There's two more principles important in this context however:
> 
> * Ordering by necessity - When given the task "send me an MP3", you near-necessarily will engage with the name, folder and format options, so these go first and in that order. all the other stuff, virtue of being further down the list, feels unimportant and "it's probably fine to keep this default", which is true.
> * Grouping by subject - When an option spawns more options (as the format dropdown and export multiple does), the "child" options want to be directly underneath the originally selected option.

@LWinterberg . Thanks for your detailed explanation.
However I think there's a problem with your ordering by necessity when it comes to the export range. Its value is not the default, it's the last used. So if the last used was current selection, and the user forgets this, because it's some way down the dialog, then they will send someone an mp3 with only some of the audio.

So I think there is case to be made for the export dialog to have the structure:
1. Export range.
2. Specification of file names, folder, and format, which depends on the export range.
3. Audio Options which depends of the format.

This would be similar to the first of the two alternative dialogs that you gave, except that:
1. In the file names, folder and format section, the format should be last so that the audio options appear immediately below.
2. There is no File name text box, as this is ignored (as it is in the current dialog).
3. The Browse button opens a dialog for selecting a folder, rather than a Save as dialog (as it is in the current dialog).

See for example the Consolidate tracks in Reaper:
![image](https://github.com/audacity/audacity/assets/11887021/bdeafb45-fcfe-4c6c-8a18-2bef58ad487a)

**Comment by petersampsonaudacity** [comment]
@LWinterberg  a mockup of what @DavidBailes  is asking for:

![image](https://github.com/audacity/audacity/assets/50205984/d44523ba-beda-43db-87b4-fe8919923938)

This looks to me as though this could work.
